.github/workflows/qodana-cloud.yaml

```diff
@@ -0,0 +1,30 @@
+name: Qodana-Cloud-Scan
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *'
+  push:
+    branches:
+      - master
+
+permissions:
+  contents: read
+jobs:
+  qodana:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+      checks: write
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+      - name: 'Qodana Scan'
+        uses: JetBrains/qodana-action@77f0ff0c702065648df9fd0340a48919dca5a1ff # v2023.2.1
+        env:
+          QODANA_TOKEN: ${{ secrets.QODANA_TOKEN }}
+        with:
+          pr-mode: false
+          post-pr-comment: false
+          args: --cleanup, -l, jetbrains/qodana-jvm:2023.2
```

Comment:
We use the enterprise scanner here because it needs a Qodana token, which is a secret. We can't use this for the PR workflow as PRs don't have access to the secret.